doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,193 @@
+plt.axvline(x=170)
+plt.savefig("brain_slice.png")
+plt.close()
+
+"""
+The marked point in the image shows a section containing cerebral spinal fluid (CSF)
+so it should have a very high f and D*, the area between the right and left is white
+matter so that should be lower, and the region on the right is gray matter and CSF.
+That should give us some contrast to see the values varying across the regions.
+Let us now take a section of the slice for fitting the IVIM model.
+"""
+
+x1, x2 = 160, 180
+y1, y2 = 90, 110
+
+plt.imshow(data[x1:x2, y1:y2, z, b].T, origin='lower', cmap="gray")
+plt.savefig("CSF_slice.png")
+plt.close()
+
+"""
+Now that we have prepared the datasets we can go forward with
+the ivim fit. Instead of fitting the entire volume, we will fit
+for the slice of data containing CSF. (x1, x2), (y1, y2) and z
+specify the coordindates of the section.
+"""
+
+data_slice = data[x1:x2, y1:y2, z, :]
+
+"""
+First, we instantiate the Ivim model in the following way.
+"""
+
+ivimmodel = IvimModel(gtab)
+
+"""
+Fitting the data is very simple. We just need to call the fit method of the
+IvimModel.
+"""
+
+ivimfit = ivimmodel.fit(data_slice, fit_method="two_stage")
+
+"""
+The fit method creates a IvimFit object which contains the
+fitting parameters of the model and can be easily extracted
+by the model_params attribute. Parameters are packed as an array
+with the last dimension being 4 which gives the (S0, f, D* , D)
+for each voxel.
+"""
+
+ivimparams = ivimfit.model_params
+print("ivimparams.shape : {}".format(ivimparams.shape))
+
+"""
+As we see, we have a 20x20 slice at the height z = 33. Thus we
+have 400 voxels. We will now plot the values of S0, f, D* and D
+for some voxels and also the various maps for the entire slice.
+This will give us an idea about the diffusion and perfusion in
+the section. Let(i, j) denote the coordinate of the voxel.
+"""
+
+i, j = 10, 10
+estimated_params = ivimfit.model_params[i, j, :]
+print(estimated_params)
+
+"""
+Let us define a plotting function. For this we will use the
+ivim_function defined in the module ivim which takes bvalues
+and ivim parameters and returns the estimated signal.
+"""
+
+
+def plot_signals(bvals, signal, estimated_params):
+    estimated_signal = ivim_function(estimated_params, bvals)
+
+    plt.scatter(bvals, signal, color="green", label="Actual signal")
+    plt.scatter(bvals, estimated_signal, color="red", label="Estimated Signal")
+    plt.xlabel("bvalues")
+    plt.ylabel("Signals")
+
+    S0_est, f_est, D_star_est, D_est = estimated_params
+    text_fit = """Estimated \n S0={:06.3f} f={:06.4f}\n
+                D*={:06.5f} D={:06.5f}""".format(S0_est, f_est, D_star_est, D_est)
+
+    plt.text(0.65, 0.50, text_fit, horizontalalignment='center',
+             verticalalignment='center', transform=plt.gca().transAxes)
+    plt.legend(loc='upper left')
+    plt.savefig(text_fit + ".png")
+    plt.close()
+
+plot_signals(gtab.bvals, data_slice[i, j, :], estimated_params)
```

Comment:
I think that I understand why you want to define this function, but since it's only called once, could we avoid defining and calling it? Instead, do all the commands in the function in the script?